src/main/java/seedu/parser/Parser.java

```diff
@@ -1,21 +1,109 @@
+        String[] tokens = parameters.split(" ");
+        int index = -1;
+        for (String token: tokens) {
+            switch (token.substring(0, 2)) {
+            case "i/":
+                index = Integer.parseInt(parameters.substring(2));
+            }
+        }
 
+        if (index == -1) {
+            throw new DukeException("index not found");
+        }
+        return index;
+    }
+
+    // TODO: [r/FREQ[/TIME or /DAY]]
     /**
-     * Get Command word from userInput.
-     *
-     * @param userInput The userInput read by Ui.
-     * @return The Command word.
+     * Parse parameters based on the following format:
+     * n/EVENTNAME [t/EVENTTIME] [d/EVENTDATE] [v/EVENTVENUE].
+     * @param parameters original parameters
+     * @return an Event object with the relevant information
      */
-    public static String getCommandWord(String userInput) {
-        return userInput.strip().split(" ")[0];
+    public Event parseEvent(String parameters) {
+        String[] tokens = parameters.split(" ");
+        String name = "";
+        String time = "";
+        String date = "";
+        String venue = "";
+
+        splitByEventFlags(tokens, name, time, date, venue);
+
+        String datetime = date + " " + time;
+
+        return new Event(name, datetime, venue);
+    }
+
+    private void splitByEventFlags(String[] tokens, String name,
+                                   String time, String date, String venue) {
+        String mostRecent = null;
+        for (String token : tokens) {
+            switch (token.substring(0, 2)) {
+            case "n/":
+                if (!name.isEmpty()) {
+                    throw new DuplicateFormatFlagsException("name");
+                }
+                name += token.substring(2);
+                mostRecent = name;
+                break;
+            case "t/":
+                if (!time.isEmpty()) {
+                    throw new DuplicateFormatFlagsException("time");
+                }
+                time += token.substring(2);
+                mostRecent = time;
+                break;
+            case "d/":
+                if (!date.isEmpty()) {
+                    throw new DuplicateFormatFlagsException("date");
+                }
+                date += token.substring(2);
+                mostRecent = date;
+                break;
+            case "v/":
+                if (!venue.isEmpty()) {
+                    throw new DuplicateFormatFlagsException("venue");
+                }
+                venue += token.substring(2);
+                mostRecent = venue;
+                break;
+            default:
+                // assumes that all valid flags have been processed before this line
+                if (isUnknownFlag(token)) {
+                    throw new UnknownFormatFlagsException("unknown flag");
+                }
+                if (mostRecent == null) {
+                    throw new NullPointerException("parameter without flag");
```

Comment:
is it better to throw all exceptions and DukeExceptions and use try catch blocks for specific exception? 
ie. catch(NullPointerException e){
      throw new DukeException(e.getMessage());
  }

Reply:
Ya, but I let all of them throw DukeException instead. I have also refactored the checks into individual functions to improve readability.